### PR TITLE
cluster up: set DNS bind and IP address for newer server versions

### PR DESCRIPTION
**Pull request by csrwng**
For origin versions >= 1.5, explicitly set the DNS bind and IP address in the node config.

### CODE REVIEW

**Comment by csrwng** [comment]
@smarterclayton ptal
tested on newer and older versions and we do the right thing

**Comment by csrwng** [comment]
[test]

File: pkg/bootstrap/docker/openshift/helper.go [hunk 1]
```diff
@@ -686,8 +718,17 @@ func (h *Helper) updateConfig(configDir string, opt *StartOptions) error {
...
 	}
-	nodeCfg.DNSBindAddress = ""
-	nodeCfg.DNSIP = ""
+	version, err := h.serverVersion()
+	if err != nil {
+		return err
+	}
+	if useBindAddress(version) {
+		nodeCfg.DNSBindAddress = "127.0.0.1:53"
+		nodeCfg.DNSIP = "172.30.0.1"
```
Comment (smarterclayton) [inline pkg/bootstrap/docker/openshift/helper.go]: this is really highlighting how problematic the "don't embed an IP" code is.  The dns ip should point to the bind address, which then ensures that the code in the node dns is used.  That is the long term flow and avoids having to query etcd for endpoints.  If we aren't going to use that, we shouldn't start it (don't pass bind address).  What happened when only dns ip was set to the service ip?

Comment (csrwng) [inline pkg/bootstrap/docker/openshift/helper.go]: So just tried... if I leave the bind address blank and only set the dns ip, then I get the right behavior:

```
oc rsh simple-ruby-1-48pcf cat /etc/resolv.conf
search default.svc.cluster.local svc.cluster.local cluster.local
nameserver 172.30.0.1
nameserver 192.168.65.1
options ndots:5
```

I'll change the code to not set the bind address.

Comment (csrwng) [inline pkg/bootstrap/docker/openshift/helper.go]: and I don't have to check the version, since the previous versions just ignore the DNSIP

Comment (csrwng) [inline pkg/bootstrap/docker/openshift/helper.go]: nm, the previous versions ignore DNSBindAddress ... not DNSIP

**Comment by csrwng** [comment]
updated the code to not set a bind address

**Comment by csrwng** [comment]
bump

**Comment by smarterclayton** [comment]
Lgtm merge at will

On Apr 1, 2017, at 4:24 PM, Cesar Wong <notifications@github.com> wrote:

bump

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/13539#issuecomment-290923171>,
or mute
the thread

**Comment by csrwng** [comment]
[merge]

**Pull request merged**